BacnetAsyncResult.cs

```diff
@@ -169,6 +169,9 @@ public TResult GetResult<TResult>(TimeSpan? timeout = null, int retryCount = 0,
                 if (retryCount == 0)
                     break;
 
+                if (Error != null)
+                    throw Error;
```

Comment:
I agree. How about in the same change you create Exception classes dedicated to the exceptions created by OnAbort OnReject and OnError so that we can forward some additional data? Should all the cases cause retransmit skipping? @DarkStarDS9  what do you think?

Reply:
It seems a little tricky to me: There are all kind of errors that can happen (`Error` is also set in catch-blocks), but only those errors that are reported by the device should not trigger a resend, IMHO.

So we probably should implement custom Exception Types and only break out of the loop if the Type matches/inherits from a remote-device reported error.

Reply:
> Should all the cases cause retransmit skipping?

I believe so, the Number_Of_APDU_Retries in the Device Object is tied to the APDU_Timeout and APDU_Segment_Timeout. Please correct me if I am wrong.

> How about in the same change you create Exception classes dedicated to the exceptions created by OnAbort OnReject and OnError so that we can forward some additional data?

Yeah I was thinking about this too, will do.

Reply:
@DarkStarDS9 Thanks that makes sense. I'll do the custom exception and the check for those before skipping.